packages/web-features-figuring/src/baseline/core-browser-set.ts

```diff
@@ -0,0 +1,44 @@
+import { Temporal } from "@js-temporal/polyfill";
+
+import {
+  BASELINE_LOW_TO_HIGH_DURATION,
+  VERY_FAR_FUTURE_DATE,
+} from "../constants";
+import { Compat } from "../browser-compat-data/compat";
+import { Release } from "../browser-compat-data/release";
+
+export const identifiers = [
+  "chrome",
+  "chrome_android",
+  "edge",
+  "firefox",
+  "firefox_android",
+  "safari",
+  "safari_ios",
+];
+
+export function browsers(compat: Compat) {
+  return identifiers.map((b) => compat.browser(b));
+}
+
+export function lowReleases(compat: Compat) {
+  return browsers(compat).map((b) => b.current());
+}
+
+export function highReleases(compat: Compat) {
+  return browsers(compat)
+    .flatMap((b) => b.releases())
+    .filter(isBaselineHighRelease);
+}
+
+function isBaselineHighRelease(release: Release) {
+  const baselineHighCutoff = Temporal.Now.plainDateISO().subtract(
+    BASELINE_LOW_TO_HIGH_DURATION,
+  );
+  return (
+    Temporal.PlainDate.compare(
+      release.date() ?? VERY_FAR_FUTURE_DATE,
+      baselineHighCutoff,
+    ) < 1
+  );
```

Comment:
Would this be equivalent?

```suggestion
  return release.date() && Temporal.PlainDate.compare(release.date(), baselineHighCutoff) <= 0;
```

Or better still, add `if (!release.date()) return false;` at the top of the function and simplify more.

Reply:
Two good points. Getting the type checking to play nice required some small adjustments. Done in 62113328b75a3827fa29856be799d7513374c98c.